.github/workflows/3-step.yml

```diff
@@ -0,0 +1,161 @@
+name: Step 3 # Connecting to GitHub
+
+on:
+  issue_comment:
+    types: [created, edited]
+
+permissions:
+  contents: read
+  actions: write
+  issues: write
+
+env:
+  STEP_4_FILE: ".github/steps/4-step.md"
+
+jobs:
+  check_keywords:
+    name: Check issue comment text for required keyword
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check for professortocat
+        id: check_professortocat
+        uses: skills/action-keyphrase-checker@v1
+        with:
+          text: ${{ github.event.comment.body }}
+          keyphrase: "professortocat"
+          case-sensitive: false
+          minimum-occurrences: 1
+      - name: Check for GitHub App url
+        id: check_alert
+        uses: skills/action-keyphrase-checker@v1
+        with:
+          text: ${{ github.event.comment.body }}
+          keyphrase: "https://github.com/apps"
+          case-sensitive: false
+          minimum-occurrences: 1
+
+  find_exercise:
+    name: Find Exercise Issue
+    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.7.0
+
+  check_step_work:
+    name: Check step work
+    runs-on: ubuntu-latest
+    needs: [check_keywords, find_exercise]
+    env:
+      ISSUE_REPOSITORY: ${{ github.repository }}
+      ISSUE_NUMBER: ${{ needs.find_exercise.outputs.issue-number }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get response templates
+        uses: actions/checkout@v4
+        with:
+          repository: skills/exercise-toolkit
+          path: exercise-toolkit
+          ref: v0.7.0
+
+      - name: Update comment - checking work
```

Comment:
Same as in previous step, this should be an "Update" as your `name` suggests but you are creating a new comment.